src/agent/Cargo.toml

```diff
@@ -71,10 +71,10 @@ openssl = { version = "0.10.38", features = ["vendored"] }
 
 # Image pull/decrypt
 [target.'cfg(target_arch = "s390x")'.dependencies]
-image-rs = { git = "https://github.com/confidential-containers/image-rs", rev = "v0.4.0", default-features = false, features = ["kata-cc-s390x"] }
+image-rs = { git = "https://github.com/confidential-containers/image-rs", rev = "756d4a4", default-features = false, features = ["encryption", "snapshot-overlayfs", "signature-simple", "getresource", "keywrap-ttrpc"] }
```

Comment:
Do you really want to go to a hash-based version here?
Also, I find the feature changes confusing. Why drop kata-cc-s390x?

Reply:
This is because now the `"kata-cc-s390"` feature equals to `["encryption", "keywrap-grpc", "snapshot-overlayfs", "signature-simple", "getresource"]`, where `"keywrap-grpc"` is enabled already.  `"keywrap-grpc"` and `"keywrap-ttrpc"` are inclusive features now.

I understand your concern. Maybe we can have a fix in `image-rs` to replace `"keywrap-grpc"` with `"keywrap-ttrpc"` in `"kata-cc-s390"`

Reply:
Hmm. That seems confusing.
The s390x platform is somewhat different, and I find it much clearer to have something that tells us explicitly "this is s390x-aware" than having the details of how it's done. I think the idea is to contrast that to regular `kata-cc`, right?

Reply:
This confusing `s390x` feature origins from this issue https://github.com/kata-containers/kata-containers/issues/5582  To put it simply this feature only controls whether `signature-cosign` is enabled. To make it transparent for kata-agent, a `s390x` feature is used which performs as a temporary solusion.

BTW, I'm working on the chain of dependencies of cosign to make it built-able on `s390x`. In future we can combine the two different features into one

Reply:
I'm afraid you misunderstood what confused me. I find 's390x' to be relatively clear, because it expresses an intent, it answers a "why?" question. By contrast, elaborating the list of sub-features hides that intent, it answers a "what?" question nobody really cares about.

To illustrate why I find this confusing, consider if I replaced:
```
   async fn pull_image(&self, req: &image::PullImageRequest) -> Result<String> {
```

with

```
   async fn set_env_var_then_setup_https_proxy_then_build_cid_from_request_then_unpack_pause_image_if_necessary_otherwise_fetch_aa_kbc_params_and_so_on(&self, req: &image::PullImageRequest) -> Result<String> {
```

I suspect you would agree that the second name is confusing, no? 😄

Reply:
Aha, you're right. Sorry for not explaining myself clearly. :(

> Maybe we can have a fix in `image-rs` to replace `"keywrap-grpc"` with `"keywrap-ttrpc"` in `"kata-cc-s390"`

1. A pr in image-rs should be patched first to include enough "nobody-really-cares-about" features for `"kata-cc-s390x"` and `"kata-cc"`
2. Then let me continue on this PR, use previous `"kata-cc-s390x"` and `"kata-cc"`
3. When no need to distinguish `"kata-cc-s390x"` and `"kata-cc"` in image-rs, we will delete feature `"kata-cc-s390x"` in image-rs and then in `kata-agent`

Is this clear and does it make sense? :P

Reply:
Yes, it makes total sense. I don't think you need a separate PR. This could be preparatory commits in this PR, if that's easier.

Reply:
Or it could be a separate PR, whatever is easier for you.